index.bs

```diff
@@ -3631,6 +3634,377 @@ opened</dfn> steps given |window|, |type| and |message|.
+of the header.
+
+<div algorithm>
+
+To <dfn>get a Header</dfn> given |name bytes| and |value bytes|:
+
+1. Let |name| be the result of [=UTF-8 decode=] with |name bytes|.
+
+   Assert: Since header names are constrained to be ASCII-only this cannot fail.
+
+1. Let |utf8 decoded value| be the result of [=UTF-8 decode without BOM or fail=]
+   with |value byes|.
+
+1. If |utf8 decoded value| is failure, then:
+
+    1. let |value| be null and |binary value| be a list.
+
+    1. For each |byte| in |value bytes|:
+
+      1. Append the [=value=] of |byte| to |binary value|.
+
+   Otherwise: let |value| be |utf8 decoded value| and let |binary value|
+   be null.
+
+1. Return a map matching the <code>NetworkHeader</code> production, with the
+   <code>name</code> field set to |string name|, the <code>value</code> field
+   set to |value| if it's not null, or omitted otherwise, and the
+   <code>binaryValue</code> field set to |binary value| if it's not null, or
+   omitted otherwise.
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkInitiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: uint,
+    ?lineNumber: uint,
+    ?stackTrace: StackTrace,
+    ?request: NetworkRequest
+};
+</pre>
+
+The <code>Initiatior</code> type represents the source of a network request.
+
+TODO: Align more closely with Fetch here?
+
+#### The network.Request type #### {#type-network-RequestId}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+NetworkRequest = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a string
+uniquely identifying that request.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkRequestData = {
+    url: text,
+    method: text,
+    headers: [ *NetworkHeader ],
+    cookies: [ *NetworkCookie ],
+    headersSize: uint,
+    bodySize: uint,
+};
+</pre>
+
+The <code>RequestData</code> type represents an ongoing network request.
+
+<div algorithm>
+
+To <dfn>get the request data</div> given |request|:
+
+1. Let |url| be the result of running the [=URL serializer=] with |request|'s
+   [=URL=].
+
+1. Let |method| be |request|'s [=method=].
+
+1. let |body size| be 0.
+
+1. Let |body| be request's [=body=].
+
+1. If |body| is a [=byte sequence=], let |body size| be the length of that
+   sequence. Otherwise if |body| is a [=body=] then let |body size| be that
+   body's [=length=] if it's not null, or 0 otherwise.
+
+   TODO: Should we handle the case where the body length is not yet known?
+
+1. Let |headers size| be the size in bytes of |request|'s [=headers list=] when
+   serialized as mandated by [=HTTP=].
+
+   Note: for protocols which allow header compression, this is the compressed
```

Comment:
I don't think this note follows from the above. If not for this note I'd assume that one should use a canonical and minimal representation, and then compress it.

Capitalize:

```suggestion
   Note: For protocols which allow header compression, this is the compressed
```